module_utils/certificate/providers/base.py

```diff
@@ -0,0 +1,291 @@
+import os
+
+from abc import ABCMeta, abstractmethod
+
+from cryptography import x509
```

Comment:
Is there a dependency on the rpm package that provides this python library?

Reply:
```
scampos@cetus:~ $ repoquery --requires --resolve python3-cryptography
Last metadata expiration check: 0:00:05 ago on Fri 15 May 2020 04:05:41 PM -03.
glibc-0:2.31-2.fc32.i686
glibc-0:2.31-2.fc32.x86_64
openssl-libs-1:1.1.1g-1.fc32.i686
openssl-libs-1:1.1.1g-1.fc32.x86_64
python3-0:3.8.2-2.fc32.i686
python3-0:3.8.2-2.fc32.x86_64
python3-cffi-0:1.14.0-1.fc32.x86_64
python3-idna-0:2.8-6.fc32.noarch
python3-six-0:1.14.0-2.fc32.noarch
```

Reply:
This code is run on the remote machine - which means it will be using python2 on a rhel7 machine.  I just wanted to make you aware that you may have to do something like - vars/certmonger-RedHat-7.yml
```yaml
__certificate_certmonger_packages: ['certmonger', 'python-cryptography']
```
to ensure the package is available on the remote machine

Reply:
Correct. The python-cryptography will actually be needed by any provider and on centos/rhel 8 the package will be python3-cryptography. I'll fix that with the changes in the `tasks/main.yml`